src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -37,6 +46,270 @@
 
 static char * PartitionBound(Oid partitionId);
 static Relation try_relation_open_nolock(Oid relationId);
+static List * CreateFixPartitionConstraintsTaskList(Oid relationId);
+static List * WorkerFixPartitionConstraintCommandList(Oid relationId, uint64 shardId,
+													  List *checkConstraintList);
+static List * CheckConstraintNameListForRelation(Oid relationId);
+static bool RelationHasConstraint(Oid relationId, char *constraintName);
+static char * RenameConstraintCommand(Oid relationId, char *constraintName,
+									  char *newConstraintName);
+
+
+PG_FUNCTION_INFO_V1(fix_partition_constraints);
+PG_FUNCTION_INFO_V1(worker_fix_partition_constraints);
+
+
+/*
+ * fix_partition_constraints fixes the constraint names of partitioned table shards on
+ * workers.
+ *
+ * Constraint names for partitioned table shards should have shardId suffixes if and only
+ * if they are unique or foreign key constraints. We mistakenly appended shardIds to
+ * constraint names on ALTER TABLE dist_part_table ADD CONSTRAINT .. queries prior to
+ * Citus 10. fix_partition_constraints determines if this is the case, and renames
+ * constraints back to their original names on shards.
+ */
+Datum
+fix_partition_constraints(PG_FUNCTION_ARGS)
```

Comment:
I'm wondering if this name can be a little more descriptive (in the context of all of PG)

e.g. fix_pre_citus10_partitioned_table_constraint_names, fix_old_distributed_partitioned_table_constraint_names

Reply:
I changed the name of the UDF to `fix_pre_citus10_partitioned_table_constraint_names` in a separate commit.